bigtable/benchmarks/benchmark.h

```diff
@@ -0,0 +1,139 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_BIGTABLE_BENCHMARKS_BENCHMARK_H_
+#define GOOGLE_CLOUD_CPP_BIGTABLE_BENCHMARKS_BENCHMARK_H_
+
+#include "bigtable/benchmarks/embedded_server.h"
+#include "bigtable/benchmarks/random.h"
+#include "bigtable/benchmarks/setup.h"
+
+namespace bigtable {
+namespace benchmarks {
+/// The result of a single operation.
+struct OperationResult {
+  bool successful;
+  std::chrono::microseconds latency;
+};
+
+struct BenchmarkResult {
+  std::chrono::milliseconds elapsed;
+  std::vector<OperationResult> operations;
+  long row_count;
+};
+
+/**
+ * Common code used by the Cloud Bigtable C++ Client benchmarks.
+ */
+class Benchmark {
+ public:
+  explicit Benchmark(BenchmarkSetup const& setup);
+  ~Benchmark();
+
+  Benchmark(Benchmark const&) = delete;
+  Benchmark& operator=(Benchmark const&) = delete;
+
+  /// Create a table for the benchmark, return the table_id.
```

Comment:
Any specific reason for using '///' instead of '//' ?

Reply:
Yes, `///` comments are used by Doxygen to generate the [documentation](https://GoogleCloudPlatform.github.io/google-cloud-cpp/), they work like `/** ... */` blocks for one-liners.